### DISCUSSION TITLE
error in runing privategpt.py

### DISCUSSION THREAD

**Discussion by dadanrhem** [imartinez/privateGPT#512] [General]
os: windows 10 64bit , ram 8gb

PrivateGPT install completed
ingest.py work well.

after running python privategpt.py i got below error.

F:\pgpt>python privategpt.py
Using embedded DuckDB with persistence: data will be stored in: db
Found model file.
gptj_model_load: loading model from 'models/ggml-gpt4all-j-v1.3-groovy.bin' - please wait ...
gptj_model_load: n_vocab = 50400
gptj_model_load: n_ctx   = 2048
gptj_model_load: n_embd  = 4096
gptj_model_load: n_head  = 16
gptj_model_load: n_layer = 28
gptj_model_load: n_rot   = 64
gptj_model_load: f16     = 2
gptj_model_load: ggml ctx size = 5401.45 MB
Traceback (most recent call last):
  File "F:\pgpt\privategpt.py", line 76, in <module>
    main()
  File "F:\pgpt\privategpt.py", line 36, in main
    llm = GPT4All(model=model_path, n_ctx=model_n_ctx, backend='gptj', callbacks=callbacks, verbose=False)
  File "pydantic\main.py", line 339, in pydantic.main.BaseModel.__init__
  File "pydantic\main.py", line 1102, in pydantic.main.validate_model
  File "C:\Python\lib\site-packages\langchain\llms\gpt4all.py", line 139, in validate_environment
    values["client"] = GPT4AllModel(
  File "C:\Python\lib\site-packages\gpt4all\gpt4all.py", line 49, in __init__
    self.model.load_model(model_dest)
  File "C:\Python\lib\site-packages\gpt4all\pyllmodel.py", line 141, in load_model
    llmodel.llmodel_loadModel(self.model, model_path.encode('utf-8'))
OSError: [WinError -1073741795] Windows Error 0xc000001d

Any one have idea what is wrong , what do I do ?

Thanks a lot

**Comment by williamshoap** [comment]
I am having the same type of issue as well. 

![image](https://github.com/imartinez/privateGPT/assets/82299588/c70d4fd1-a285-40fa-a0f3-646fbd94756f)

**Comment by styck** [comment]
I get the same exact error, I believe it is in the GPT4ALL package for windows, maybe a dependency in one of the DLL's?  I got the PrivateGPT code working in Google Colab so don't think the problem is there, just the packages that we use for Windows. Maybe related to this: https://github.com/nomic-ai/gpt4all/issues/717, there are some new comments there since yesterday so I'll try again later.

**Reply by progone** [reply]
having the same issue on Macbook Pro (intel CPU)

**Comment by dadanrhem** [comment]
pip3 cache purge 
pip3 uninstall gpt4all
pip3 install gpt4all==.0.2.2

even with version 0.2.2 gpt4all I got same error.

**Reply by progone** [reply]
I just tried this solution. It failed . (Macbook Pro (intel CPU))
`Desktop/privateGPT/privateGPT.py", line 36, in main
    llm = GPT4All(model=model_path, n_ctx=model_n_ctx, backend='gptj', callbacks=callbacks, verbose=False)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for GPT4All
__root__
  Failed to retrieve model (type=value_error)`

**Comment by jkhalil01** [comment]
I get the same error on windows 10

**Comment by williamshoap** [comment]
Yeah, I think it's safe to say that it's broke

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by dadanrhem** [comment]
https://github.com/nomic-ai/gpt4all/issues/758
join here for more investigation

**Comment by cosmic-snow** [comment]
By the way, if anyone is still following this: It was ultimately resolved in the above mentioned issue in the GPT4All project.

The problem was that the CPU didn't support the AVX2 instruction set. To check that, you could e.g. use the HWINFO tool. A CPU with AVX2 support looks something like this:
![CPU supports AVX2](https://user-images.githubusercontent.com/134004613/241822981-e09b3667-6bcf-43f4-ad29-76b7db16bd79.png)

If yours doesn't, the solution is to compile the DLLs for the bindings yourself. We went through that in the above issue and there's a summary at the very bottom. Maybe start with that to see if you want to attempt that yourself.

**Reply by cosmic-snow** [reply]
Note: This is already outdated. The GPT4All main branch now builds multiple libraries. If you build from the latest, "AVX only" isn't a build option anymore but should (hopefully) be recognised at runtime.

Once these changes make their way into a PyPI package, you likely won't have to build anything anymore, either. (fingers crossed)

**Reply by cosmic-snow** [reply]
PyPI package was updated.

I don't know if it's already ok here, but once this project is compatible: try `pip install -U gpt4all` instead of building yourself. It should already include the 'AVX only' build in a DLL and use that automatically if your hardware doesn't support AVX2.

**Reply by cosmic-snow** [reply]
PyPI package was updated again (v0.3.3 now). Try that.

**Reply by cosmic-snow** [reply]
PyPI package updated again (v0.3.4), fixes an issue with `hermes` and maybe other newer ones.

**Comment by thanhtantran** [comment]
I even have this weird error.

```
ubuntu@webserver:~/privateGPT$ python privateGPT.py
Using embedded DuckDB with persistence: data will be stored in: db
Found model file.
gptj_model_load: loading model from 'models/ggml-gpt4all-j-v1.3-groovy.bin' - please wait ...
gptj_model_load: n_vocab = 50400
gptj_model_load: n_ctx   = 2048
gptj_model_load: n_embd  = 4096
gptj_model_load: n_head  = 16
gptj_model_load: n_layer = 28
gptj_model_load: n_rot   = 64
gptj_model_load: f16     = 2
Illegal instruction (core dumped)
```

note: installed successfully, ingest runs well

**Reply by mibtim** [reply]
hi mate, i got the same error. i found this forum it appears possibly due to a running on vm. to be not sure if that's the case. https://github.com/imartinez/privateGPT/issues/326... if you happen to find the solution ps let me also know. thanks

**Reply by cosmic-snow** [reply]
'Illegal instruction' is probably caused by your CPU not supporting AVX 2, as said in my previous comment.

Builds that support running on that kind of hardware are technically already included in the PyPI package now, but it seems there are still a few bugs with the whole thing.

**Reply by mibtim** [reply]
hi mate, thanks for the reply. as i'm running on windows 10 (Intel(R) Core(TM) i7 CPU @ 2.20GHz   2.21 GHz) vm continue search for a resolution to get this working. manage to find useful info on this article and as it got to do with windows security relate not a bug.. i followed the instructions and it worked for me. not sure if this helps u but worth the try. 
https://stackoverflow.com/questions/65780506/how-to-enable-avx-avx2-in-virtualbox-6-1-16-with-ubuntu-20-04-64bit

**Reply by cosmic-snow** [reply]
Thanks, but I don't have that issue myself. I *am* running VirtualBox, but already had Hyper-V disabled. Good info for others, though.

Although it might even come in handy for troubleshooting someone else's problem at some point.

**Reply by mibtim** [reply]
cool hope u find a solution. good luck.
